src/seedu/addressbook/storage/StorageFile.java

```diff
@@ -106,7 +106,9 @@ public void save(AddressBook addressBook) throws StorageOperationException {
             marshaller.marshal(toSave, fileWriter);
 
         } catch (IOException ioe) {
-            throw new StorageOperationException("Error writing to file: " + path);
+            if (!Files.isWritable(path)) {
```

Comment:
Try to eliminate this explicit check. You are "forcing" an exception check here. 
If you are doing this, you might as well handle the situation in your code. :smile: